fetcher/src/job/job.utils.ts

```diff
@@ -92,23 +117,55 @@ function validateAdapter(adapter): IAdapter {
   }
 }
 
+function selectProxy(url: string, proxies) {
+  const source = url.split('/')[2]
```

Comment:
This is very error prone approach to getting host from url. Please consider the solution below.

```typescript
const source = (new URL(url)).host
```